src/libs/Fullstory/index.ts

```diff
@@ -91,5 +101,86 @@ const FS = {
     init: (_value: OnyxEntry<UserMetadata>) => {},
 };
 
+/**
+ * Extract values from non-scraped at build time attribute WEB_PROP_ATTR,
+ * reevaluate "fs-class".
+ */
+function parseFSAttributes(): void {
```

Comment:
I’m concerned about this function; it seems to be executed multiple times in a row. Each execution tests approximately 300 elements, and this occurs when switching reports, even if switching between the same reports.

<img width="237" alt="Screenshot 2024-11-06 at 19 11 58" src="https://github.com/user-attachments/assets/79a035e2-9a88-400e-8bde-6d54275b3a72">

Would be great to avoid it, given that I felt the report switching lagging

Reply:
I also suggest to add test on this function, which seems complex and any future update should be cover by tests